src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -827,12 +833,24 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
     notExpected.addAll(actualEntries.keySet());
   }
 
+  private <V, K> Map<K, V> instantiate(Class<? extends Map> mapClass) {
+    try {
+      Constructor<?> constructor = mapClass.getConstructor();
+      return (Map<K, V>) constructor.newInstance();
+      // return objenesis.newInstance(class1);
```

Comment:
I have setup objenesis, uncomment to see the error we have when using it

Reply:
We should create an `InstantiatorStrategy` with different implementation instead of this basic approach.

Reply:
Just a thought - perhaps trying a `clone()` if the class implements `Cloneable` could be one of the implementations of `InstantiatorStrategy`.  `java.util.HashMap` implements this, so it's likely to be common